friktion/friktion/bid_details.py

```diff
@@ -6,30 +6,43 @@
 from solana.publickey import PublicKey
 from solders.signature import Signature
 
+from friktion.pda import find_swap_order_address
+
 
 @dataclass
 class BidDetails:
     signer_wallet: PublicKey
+    # necessary to verify swap order public key
+    creator: PublicKey
     order_id: int
     bid_size: int  # 1e9
     bid_price: int  # 1e9
     referrer: PublicKey
 
-    def as_msg(self):
+    def as_msg(self) -> bytes:
         give_amount = self.bid_size
         receive_amount = self.bid_price * self.bid_size
-        payload = [
-            [self.order_id],
-            str(self.signer_wallet),
-            str(self.referrer),
-            [give_amount, receive_amount],
-        ]
-
-        # set separators to remove whitespaces
-        dumped_payload = json.dumps(payload, separators=(',', ':'))
-
-        return bytes(dumped_payload, 'utf-8')
+        payload = bytearray()
+        payload.extend(self.order_id.to_bytes(
+            8,
+            'little',
+            signed = False
+        ))
+        swap_order_pk = find_swap_order_address(self.creator, self.order_id)[0]
+        payload.extend(self.signer_wallet.__bytes__())
+        payload.extend(self.referrer.__bytes__())
+        payload.extend(swap_order_pk.__bytes__())
+        payload.extend(give_amount.to_bytes(8,'little',
+                    signed = False
+))
+        payload.extend(receive_amount.to_bytes(
+            8,'little',
+                        signed = False
 
+        ))
```

Comment:
Also, could we provide the `swap_order_address` as a parameter instead of receiving the `creator` to pull that? 
This would save a call to the chain.

---
Going further in that direction, I am thinking that we should start providing the `swap_order_address` as the `verifying_contract` in the rfq creation endpoint.

Context: currently, the options contract address (`GriGJSF84XdPq6Td6u6Hu8oqKgwTXY94fvwJrJf1gQTW`) is being provided as the `verifying_contract`. However, it is hardcoded and is not being used to pull any info. By changing that, we would have the `swap_order_address` to build the bid details and we could also simplify the `get_offer_details` method to take that address as a parameter (saving another call to the chain).
Do you see any problems for that change?

cc: @pdonorio

Reply:
@bigdefidev is `find_swap_order_address` the main reason why we are doing this?